eth2util/signing/signing.go

```diff
@@ -42,14 +46,15 @@ const (
 	// DomainContributionAndProof        DomainName = "DOMAIN_CONTRIBUTION_AND_PROOF".
 )
 
-// Eth2DomainProvider is the subset of eth2 beacon api provider required to get a signing domain.
-type Eth2DomainProvider interface {
-	eth2client.SpecProvider
+// Eth2Provider is the subset of eth2 beacon api provider required to get a signing domain.
+type Eth2Provider interface {
```

Comment:
could split into into two interfaces, `Eth2DomainProvider` and `Eth2VerifyProvider` so not leak dependencies between the two functions.